service/worker/scanner/README.md

```diff
@@ -0,0 +1,247 @@
+at the core of these processes:
+```go
+scanner := NewScanner(
+    shardID,
+    ctx.Hooks.Iterator(activityCtx, pr, params),
+    resources.GetBlobstoreClient(),
+    params.BlobstoreFlushThreshold,
+    ctx.Hooks.Manager(activityCtx, pr, params, resources.GetDomainCache()),
+    func() { activity.RecordHeartbeat(activityCtx, heartbeatDetails) },
+    scope,
+    resources.GetDomainCache(),
+)
+```
+
+And last but not least, note that:
+- Config is saved when the workflow starts, not read from scratch when the activities begin.
+- Scanner/fixer are on an every-minute cron schedule, but are allowed to run for ludicrous amounts of time.
+- This means any config changes will _very likely_ need to wait for the current run to finish before picking up your changes.
+  - You can advance this relatively quickly by terminating the existing workflows, and then restarting a workers-service.
+  - Terminating will stop the cron completely, but restarting any instance will start a new cron, which will pick up the new config.
+
+<details><summary>Enabling scanners for local dev</summary>
+
+For local purposes, you can pretty easily enable these via the dynamicconfig
+file store [config/dynamicconfig/development.yaml](../../../config/dynamicconfig/development.yaml):
+```yaml
+# inconsistent key names, unfortunately.  don't guess, copy/paste
+worker.concreteExecutionFixerEnabled:
+- value: true
+  constraints: {}
+worker.executionsScannerInvariantCollectionStale:
+- value: true
+  constraints: {}
+worker.executionsScannerEnabled:
+- value: true
+  constraints: {}
+```
+
+</details>
+
+## ShardScanner and ShardFixer implementations
+
+Implemented in:
+- [shardscanner/scanner.go](shardscanner/scanner.go)
+- [shardscanner/fixer.go](shardscanner/fixer.go)
+
+Once again, these are fairly similar structurally:
+- Iterate over their data (from the scan-paginator or the blobstore to-fix paginator)
+- Run the invariant manager's check or fix on each item
+- Collect aggregated metrics
+- Periodically flush details to a blob-store
+
+# Some other files involved that seemed interesting to link
+
+- [/common/reconciliation/invariant/types.go](../../../common/reconciliation/invariant/types.go)
+  - `Name` entries name and refer to individual `Invariant` implementations.
+  - `Collection` entries define... categories of sorts, which are used elsewhere to configure which `Invariant`s run.
+- [/service/worker/scanner/executions/types.go](executions/types.go)
+  - `Collection` selections elsewhere are converted to `Invariant`s, based on the scanner type
+    - Timer scanners are handled in [scanner/timers/timers.go](timers/timers.go) rather than in here
+- [/service/worker/scanner/executions/concrete_execution.go](executions/concrete_execution.go)
+  - this and [current_execution.go](executions/current_execution.go) define a `[ExecutionType]ExecutionConfig` func that can override which `Collection`s are run in a scanner.
+
+# Wishlist
```

Comment:
I would just cut this into a design proposal, it's inclusion here as-is makes it look to the casual reader as if we have no control over our own destiny (which may or may not be true) but doesn't make it clear that you're making a time-tradeoff with the PR.